cdap-validation/bin/testing.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/python
+
+import test_helpers as helpers
+import json
+import ambari
+import cm
+
+##### TEST FRAMEWORK FUNCTIONS #####
+
+input_json = "input.json"
+
+# calls all other functions
+def testing(base_vars):
+    print "start testing"
+
+    # prep input variables
+    cluster_vars = {}
+    input = read_input(input_json)
+    cluster_vars['host'] = input['params']['uri']
+    cluster_vars['username'],cluster_vars['password']= input['params']['userpass'].split(':')
+    cluster_vars['verbose'] = input['params']['verbose']
+    cluster_vars['cluster'] = input['params']['cluster_name']
+    cluster_vars['modules'] = input['params']['modules']
+
+    # find_modules
+    # hold on for now
+
+    # detect_installed_services
+    # hold on for now
+
+    # prep the url
+    find_api='/api'
+    if cluster_vars['host'].find(find_api) == -1:
+        cluster_vars['host'] += 'api/'
+    print 'uri=%s' % (cluster_vars['host'])
+
+    # validate_connection
+    print "validate connection"
```

Comment:
Output should be better. Use proper tense and capitalization. It can still be terse, for now.